### PR TITLE
Azure Cluster

### PR DESCRIPTION
**Pull request by pataruco**
# Azure Kubernetes Cluster

[Airtable ticket](https://airtable.com/tbl7CISHKkr8Kdeh2/viwlDIU6cp8cbQYhj/rec3dNRwlGvNtR4li?blocks=hide)

This solves  #248 

### Acceptance Criteria

- [ ] Create an Azure cluster (AKS) (set up infrastructure)

### Testing information

NA

### Pre-flight Checklist

- ~Tests~
-  ~DUXD review approval~
- ~Accessibility Reviewed~
- ~Product owner demo~

### Pre-merge Checklist

- [ ] Branch test approved

### CODE REVIEW

File: infrastructure/README.md [hunk 1]
```diff
@@ -52,3 +59,30 @@ This folder contains all the Terraform files for provisioning infrastructure in
...
+1. [Authenticate](https://www.terraform.io/docs/providers/azurerm/guides/azure_cli.html) with Azure
+
+   ```shell
+   az login
+
+   # SUBSCRIPTION_ID can be found in output from previous command
+   az account set --subscription="SUBSCRIPTION_ID"
+   ```
+
+2. [Create a service principal](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest#password-based-authentication) password based authentication
```
Comment (craiga) [inline infrastructure/README.md]: Should this describe what `<ServicePrincipalName>` should be?

File: infrastructure/environments/non-prod/main.tf [hunk 2]
```diff
@@ -1,13 +1,16 @@
 provider "azurerm" {
-  version = "=1.38.0"
+  version = "~> 1.38.0"
+  # subscription_id = "b2f4966f-71ea-415b-9ec6-bf96d597d596"
```
Comment (craiga) [inline infrastructure/environments/non-prod/main.tf]: Why is this here? Do we need to regenerate this?

File: infrastructure/scripts/create-kubernetes-config.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+set -euo pipefail
+
+PS3='Please enter your current enviroment: '
```
Comment (craiga) [inline infrastructure/scripts/create-kubernetes-config.sh]: "environment" is missing an "n".

File: infrastructure/environments/non-prod/main.tf (hunk 2 above)
Comment (pataruco) [reply infrastructure/environments/non-prod/main.tf]: Good catch, I removed the comment

File: infrastructure/README.md (hunk 1 above)
Comment (pataruco) [reply infrastructure/README.md]: I add a coment to replace `<ServicePrincipalName>` with the name of the account you want to create

File: infrastructure/scripts/create-kubernetes-config.sh (hunk 3 above)
Comment (pataruco) [reply infrastructure/scripts/create-kubernetes-config.sh]: Thanks
